easybuild/easyblocks/f/fftw.py

```diff
@@ -0,0 +1,197 @@
+            if hasattr(self, flag):
+                raise EasyBuildError("EasyBlock attribute '%s' already exists")
+            setattr(self, flag, self.cfg['use_%s' % flag])
+
+        cpu_features = get_cpu_features()
+
+        # on macOS, AVX is indicated with 'avx1.0' rather than 'avx'
+        if 'avx1.0' in cpu_features:
+            cpu_features.append('avx')
+
+        self.log.info("List of CPU features: %s", cpu_features)
+
+        # auto-detect CPU features that can be used and are not enabled/disabled explicitly,
+        # but only if --optarch=GENERIC is not being used
+        if self.cfg['auto_detect_cpu_features'] and build_option('optarch') != OPTARCH_GENERIC:
+            for flag in FFTW_CPU_FEATURE_FLAGS:
+                if getattr(self, flag) is None and flag in cpu_features:
+                    self.log.info("Enabling use of %s (should be supported based on CPU features)", flag.upper())
+                    setattr(self, flag, True)
+
+    def run_all_steps(self, *args, **kwargs):
+        """
+        Put configure options in place for different precisions (single, double, long double, quad).
+        """
+        # keep track of configopts specified in easyconfig file, so we can include them in each iteration later
+        common_config_opts = self.cfg['configopts']
+
+        self.cfg['configopts'] = []
+
+        for prec in FFTW_PRECISION_FLAGS:
+            if self.cfg[EB_FFTW._prec_param(prec)]:
+
+                prec_configopts = []
+
+                # double precison is the default, no configure flag needed (there is no '--enable-double')
+                if prec != 'double':
+                    prec_configopts.append('--enable-%s' % prec)
+
+                # MPI is not supported for quad precision
+                if prec != 'quad-precision' and self.cfg['with_mpi']:
+                    prec_configopts.append('--enable-mpi')
+
+                if self.toolchain.options['pic']:
+                    prec_configopts.append('--with-pic')
+
+                for libtype in ['openmp', 'threads']:
+                    if self.cfg['with_%s' % libtype]:
+                        prec_configopts.append('--enable-%s' % libtype)
+
+                # SSE2, AVX* only supported for single/double precision
+                if prec in ['single', 'double']:
+                    for flag in FFTW_CPU_FEATURE_FLAGS_SINGLE_DOUBLE:
+                        if getattr(self, flag):
+                            prec_configopts.append('--enable-%s' % flag)
+
+                # Altivec (POWER) and SSE only for single precision
+                for flag in ['altivec', 'sse']:
+                    if prec == 'single' and getattr(self, flag):
+                        prec_configopts.append('--enable-%s' % flag)
+
+                # NEON (ARM) only for single precision and double precision (on AARCH64)
+                if (prec == 'single' and self.neon) or (prec == 'double' and self.asimd):
```

Comment:
The thing is: the `neon` flag is only available on AArch32 if NEON support is available. And it is single-precision only. On AArch64, there is no `neon` flag, but only the `asimd` flag indicating single- and double-precision NEON support. So yes, I believe it would make a difference.